recipes/ncbi-cxx-toolkit-public/26.0/conanfile.py

```diff
@@ -0,0 +1,1587 @@
+
+def _remove_readonly(fn, path, excinfo):
+    try:
+        os.chmod(path, stat.S_IWRITE)
+        fn(path)
+    except Exception as exc:
+        print("Skipped: ", path, " because ", exc)
+
+class NcbiCxxToolkit(ConanFile):
+    name = "ncbi-cxx-toolkit-public"
+    license = "CC0-1.0"
+    homepage = "https://ncbi.github.io/cxx-toolkit"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = "NCBI C++ Toolkit"
+    topics = ("ncbi", "biotechnology", "bioinformatics", "genbank", "gene",
+              "genome", "genetic", "sequence", "alignment", "blast",
+              "biological", "toolkit", "c++")
+    settings = "os", "compiler", "build_type", "arch"
+
+    tk_tmp_tree = ""
+    tk_src_tree = ""
+    all_NCBI_requires = {}
+
+    options = {
+        "shared":     [True, False],
+        "sharedDeps": [True, False],
+        "fPIC":       [True, False],
+        "projects":   "ANY",
+        "targets":    "ANY",
+        "tags":       "ANY"
+    }
+    default_options = {
+        "shared":     False,
+        "sharedDeps": False,
+        "fPIC":       True,
+        "projects":   "",
+        "targets":    "",
+        "tags":       ""
+    }
+    NCBI_to_Conan_requires = {
+#                Linux                  Macos               Windows
+#"BACKWARD":     ("backward-cpp/1.6",    "",                 ""),
+"BACKWARD":     (None,                  None,               None),
+"BerkeleyDB":   ("libdb/5.3.28",        "libdb/5.3.28",     None),
+#"Boost":        ("boost/1.76.0",        "boost/1.76.0",     "boost/1.76.0"),
+"BZ2":          ("bzip2/1.0.8",         "bzip2/1.0.8",      "bzip2/1.0.8"),
+"CASSANDRA":    ("cassandra-cpp-driver/2.15.3", None,       None),
+"GIF":          ("giflib/5.2.1",        "giflib/5.2.1",     "giflib/5.2.1"),
+#"GRPC":         ("grpc/1.39.1",         "grpc/1.39.1",      "grpc/1.39.1"),
+"JPEG":         ("libjpeg/9d",          "libjpeg/9d",       "libjpeg/9d"),
+"LMDB":         ("lmdb/0.9.29",         "lmdb/0.9.29",      "lmdb/0.9.29"),
+"LZO":          ("lzo/2.10",            "lzo/2.10",         "lzo/2.10"),
+"MySQL":        ("libmysqlclient/8.0.25", "libmysqlclient/8.0.25", "libmysqlclient/8.0.25"),
+"NGHTTP2":      ("libnghttp2/1.45.1",   "libnghttp2/1.45.1","libnghttp2/1.45.1"),
+"PCRE":         ("pcre/8.45",           "pcre/8.45",        "pcre/8.45"),
+"PNG":          ("libpng/1.6.37",       "libpng/1.6.37",    "libpng/1.6.37"),
+#"PROTOBUF":     ("protobuf/3.17.1",     "protobuf/3.17.1",  "protobuf/3.17.1"),
+"SQLITE3":      ("sqlite3/3.36.0",      "sqlite3/3.36.0",   "sqlite3/3.36.0"),
+"TIFF":         ("libtiff/4.3.0",       "libtiff/4.3.0",    "libtiff/4.3.0"),
+"XML":          ("libxml2/2.9.12",      "libxml2/2.9.12",   "libxml2/2.9.12"),
+"XSLT":         ("libxslt/1.1.34",      "libxslt/1.1.34",   "libxslt/1.1.34"),
+"UV":           ("libuv/1.42.0",        "libuv/1.42.0",     "libuv/1.42.0"),
+"Z":            ("zlib/1.2.11",         "zlib/1.2.11",      "zlib/1.2.11"),
```

Comment:
The mapping from NCBI to reference names is needed, but I see that most (all?) versions are the same. Probably we can just list them in a regular `requirements()` method for now. If in the future we see that things start to break we can think about something else.

Reply:
Most, but not all. For example, libdb "does not support Visual Studio 2019" (very funny indeed). When migrating from one version to another, it is sometimes convenient to use different versions on different OS.